### ISSUE TITLE
Question about calculate full hessian

### ISSUE THREAD

**Issue by BossunWang** [Animadversio/GAN-Geometry#7]
The mov_vect is same as  ref_vect, but self.perturb_vec is random in GANHVPOperator
Is the mov_vect should be perturbation when using "BP" to calculate full hessian?

https://github.com/Animadversio/GAN-Geometry/blob/a6e0e555d3c1449a72ca1758e71d4aac3364c03c/core/GAN_hessian_compute.py#L55

https://github.com/Animadversio/GAN-Geometry/blob/a6e0e555d3c1449a72ca1758e71d4aac3364c03c/core/GAN_hvp_operator.py#L74

**Comment by Animadversio** [comment]
Thanks for examining my code and noticing this difference! 

I think if we set `self.perturb_vec` as zero vector but require gradient `torch.zeros((1, self.size), dtype=torch.float32).requires_grad_(True)`, it will yield the same result. 
I think it's at the start of the project I think zero vector will create zero distance $d$, which might invoke some numerical issue when evaluating gradient, so I add a small random $dz$ just to be safe. But actually, it didn't have numerical issue. 

The BP one implement 
$$d = D(G(sg(z)), G(z))$$ 
$sg$ means stop gradient. and it evaluates $$\partial d/\partial z$$ which will have the same result as the Backword HVP case. 
$$d = D(G(z), G(z+dz))$$
and evaluating $\partial d / \partial dz$ at $dz=0$
